question-service/model/question-orm.js

```diff
@@ -1,8 +1,17 @@
-import { findQuestionByDifficulty } from './repository.js'
+import { findQuestionByFilter } from './repository.js'
 
 export const ormFindByDifficulty = async (difficulty) => {
   try {
-    return await findQuestionByDifficulty({ difficulty: difficulty })
+    return await findQuestionByFilter({ difficulty: difficulty })
+  } catch (err) {
+    console.log('ERROR: Could not retrieve question')
+    return { err }
+  }
+}
+
+export const ormFindById = async (id) => {
+  try {
+    return await findQuestionByFilter({ qnsid: id })
```

Comment:
Since qnsid is unique, we can maybe use a `findOneQuestionByFilter` function instead that uses `QuestionModel.findOne()`. Then the frontend doesn't have to access the document it needs with `[0]` everytime

Reply:
Also another question that might be more for @RyanCheungJF: the `qnsid` field doesn't seem to be in the question model schema LOL how come we have this field ah

Reply:
I think the reason why I didn't use fineOne was that when filtering by difficulty, it returns an array of questions instead. which is of the intended use case.

Reply:
Yeah, filtering by difficulty will still use `findQuestionByFilter` and `find`, but I'm thinking filtering by id will use a different function that instead uses `findOne`. Cos when filtering by the difficulty you do indeed expect an array of questions to be returned, but when filtering by id you only expect one question. Just a small suggestion though!

Reply:
ah i see! will make the difference in api call before pushing! Thanks for the insightful suggestion.